File: mobile/src/main/java/treehou/se/habit/ui/servers/create/custom/SetupServerComponent.java
package treehou.se.habit.ui.servers.create.custom;


import dagger.Subcomponent;
import treehou.se.habit.module.ActivityScope;
import treehou.se.habit.module.FragmentComponent;
import treehou.se.habit.module.FragmentComponentBuilder;

@ActivityScope
@Subcomponent(
        modules = SetupServerModule.class
)
public interface SetupServerComponent extends FragmentComponent<SetupServerFragment> {

    @Subcomponent.Builder
    interface Builder extends FragmentComponentBuilder<SetupServerModule, SetupServerComponent> {
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/servers/create/custom/SetupServerComponent.kt
package treehou.se.habit.ui.servers.create.custom


import dagger.Subcomponent
import treehou.se.habit.module.ActivityScope
import treehou.se.habit.module.FragmentComponent
import treehou.se.habit.module.FragmentComponentBuilder

@ActivityScope
@Subcomponent(modules = arrayOf(SetupServerModule::class))
interface SetupServerComponent : FragmentComponent<SetupServerFragment> {

    @Subcomponent.Builder
    interface Builder : FragmentComponentBuilder<SetupServerModule, SetupServerComponent>
}